laplace/marglik.py

```diff
@@ -0,0 +1,248 @@
+from copy import deepcopy
+import numpy as np
+import torch
+from torch.optim import Adam
+from torch.nn import CrossEntropyLoss, MSELoss
+from torch.nn.utils import parameters_to_vector
+import logging
+
+from laplace import KronLaplace
+from laplace.curvature import AsdlGGN
+from laplace.utils import expand_prior_precision
+
+
+def marglik_training(
+    model,
+    train_loader,
+    likelihood='classification',
+    optimizer_cls=Adam,
+    optimizer_kwargs=None,
+    scheduler_cls=None,
+    scheduler_kwargs=None,
+    n_epochs=300,
+    laplace=KronLaplace,
```

Comment:
Maybe it makes sense to use the `Laplace` function interface, since we use it everywhere else as the UI. So here you would pass the `laplace_kwargs` or `subset_of_weights`, `hessian_structure`, and `laplace_kwargs`.

Reply:
Done as suggested, thanks.